classes/classes/Cock.as

```diff
@@ -12,14 +12,26 @@ package classes
 		
 		private static const SERIALIZATION_VERSION:int = 1;
 		
+		private static const OBJECT_NOT_FOUND:int = -1;
+
 		public static const MAX_LENGTH:Number = 9999.9;
 		public static const MAX_THICKNESS:Number = 999.9;
+		public static const KNOTMULTIPLIER_NO_KNOT:Number = 1;
 		
 		private var _cockLength:Number;
 		private var _cockThickness:Number;		
 		private var _cockType:CockTypesEnum;	//See CockTypesEnum.as for all cock types
 		
-		//Used to determine thickness of knot relative to normal thickness
+		/**
+		 * Contains cock types that support knots.
+		 * This is used for the supportsKnot method and with that to decide if a knot is reset when the cock type changes
+		 * e.g. player.cocks[0].cockType = CockTypesEnum.HORSE
+		 */
+		private static const cockTypesWithKnots:Vector.<CockTypesEnum> = new <CockTypesEnum> [CockTypesEnum.DOG, CockTypesEnum.FOX, CockTypesEnum.WOLF, CockTypesEnum.DRAGON];
```

Comment:
I wonder(ed) if it would make sense to move that lists into the `classes.lists` section(=package): [Example diff](https://gist.github.com/Stadler76/5ddec876af794dcfc31de00dece15312). So its a bit easier to find for coders who intend to add a knotted cock type to the game.

On the other hand, I'd suggest that every list item is in a separate line and has a trailing comma (for less noisy diffs, when adding items to it later on):
```as3
private static const cockTypesWithKnots:Vector.<CockTypesEnum> = new <CockTypesEnum> [
	CockTypesEnum.DOG,
	CockTypesEnum.FOX,
	CockTypesEnum.WOLF,
	CockTypesEnum.DRAGON,
];
```
Oh and btw: Doesn't Sonarcube complain about lowercase-letters in constant names?

Well, these are just suggestions.

Reply:
Didn't know about the `list` package. Feel free to push the diff as a commit.

You are right, SonarCube would flag lowercase const.

Reply:
Ok, I'll push it once the local ant build runs through without errors.